indica/readers/adas.py

```diff
@@ -148,6 +150,119 @@ def get_adf11(self, quantity: str, element: str, year: str) -> DataArray:
             attrs=attrs,
         )
 
+    def get_adf15(self, quantity: str, element: str, charge: str) -> DataArray:
+        """Read data from the specified ADAS file.
+
+        Parameters
+        ----------
+        quantity
+            The type of data to retrieve. Options: ic, cl, ca, ls, llu.
+        element
+            The atomic symbol for the element which will be retrieved.
+        charge
+            Charge state of the ion (e.g. 16 for Ar 16+).
+
+        Returns
+        -------
+        :
+            The data in the specified file. Dimensions are density and
+            temperature. Each members of the dataset correspond to a
+            different charge state.
+
+        """
+        now = datetime.datetime.now()
+        if quantity.lower()=="llu":
+            file_component = "transport"
+        else:
+            raise NotImplementedError(
+                "{} new format 'pec40' not implemented yet.".format(self.__class__.__name__)
+            )
+            # file_component = f"pec40][{element.lower()}"
+
+        filename = ( Path(pathname2url(file_component)) /
+                     pathname2url(f"{file_component}_{quantity.lower()}][{element.lower()}{charge}.dat")
+                     )
+        with self._get_file("adf15", filename) as f:
+            header = f.readline().split()
+            ntrans = int(header[0])
+            element_name = header[1][1:3].lower()
+            assert element_name==element.lower()
+            charge_state = header[1][4:6].lower()
+            assert charge_state==charge
+
+            date = datetime.date.min
+            for i in range(ntrans):
+                section_header = ""
+                section_header = f.readline()
+                m = re.search(r"ISEL =\s*(\d+)", section_header, re.I)
+                assert isinstance(m, re.Match)
+                assert int(m.group(1))-1 == i
+
+                m = re.search(r"TYPE =\s*(\S+)", section_header, re.I)
+                pec_type = m.group(1)
+
+                section_header = section_header.split()
+                wavelength = float(section_header[0]) / 10. # (nm)
+
+                if i==0:
+                    nd = int(section_header[2])
+                    nt = int(section_header[3])
```

Comment:
Perhaps parse these before entering the loop? that way you could declare data outside of the loop as well and avoid the need for this if/else block.

Reply:
Done

Reply:
Done